In [1]:
import pandas as pd

#### joining child internet access df to child obesity df

In [16]:
access = pd.read_csv('cleaned_data/USA_internet_access_state.csv')
access['State'] = access['State'].str.strip('.')
access.head(3)

Year    State Individuals (thousands) Percent Individuals 3 - 17  \
0  2001  Alabama                    1674                     22.24%   
1  2001   Alaska                     377                     22.24%   
2  2001  Arizona                    2637                     22.24%   

   Individuals 3 - 17 (thousands)  Individuals 3 - 17  
0                             372              372391  
1                              84               83774  
2                             586              586415

In [17]:
obesity = pd.read_csv('cleaned_data/USA_child_obesity_2001_2019.csv')
obesity = obesity.drop(columns = ['Unnamed: 0'])
obesity.head(3)

YearStart          LocationDesc  \
0       2019               Arizona   
1       2019  District of Columbia   
2       2017               Vermont   

                                            Question  Data_Value  Sample_Size  \
0  Percent of students in grades 9-12 who have ob...        13.3       1739.0   
1  Percent of students in grades 9-12 who have ob...        17.1       8978.0   
2  Percent of students in grades 9-12 who have an...        14.1      18504.0   

   Total                    GeoLocation StratificationID1  
0  Total  (34.86597028, -111.763811277)           OVERALL  
1  Total  (38.890371385, -77.031961127)           OVERALL  
2  Total  (43.625381239, -72.517640791)           OVERALL

In [20]:
print('access years = ', access.Year.unique())
print('obesity years = ', obesity.YearStart.unique())

access years =  [2001 2003 2007 2009 2010 2011 2012]
obesity years =  [2019 2017 2009 2007 2011 2005 2013 2003 2001 2015]


#### obesity df does not have year 2010 or 2012. I am changing 2013 in obesity df to 2012 and will make this known in the final report.

In [23]:
for index, row in obesity.iterrows():
    if row.YearStart == 2013:
        obesity.loc[index, 'YearStart'] = 2012

In [28]:
access_obesity = pd.merge(access, obesity, 
                               left_on = ('Year', 'State'), right_on = ('YearStart', 'LocationDesc'), 
                               how = 'left')
access_obesity.head(3)

Year    State Individuals (thousands) Percent Individuals 3 - 17  \
0  2001  Alabama                    1674                     22.24%   
1  2001  Alabama                    1674                     22.24%   
2  2001   Alaska                     377                     22.24%   

   Individuals 3 - 17 (thousands)  Individuals 3 - 17  YearStart LocationDesc  \
0                             372              372391     2001.0      Alabama   
1                             372              372391     2001.0      Alabama   
2                              84               83774        NaN          NaN   

                                            Question  Data_Value  Sample_Size  \
0  Percent of students in grades 9-12 who have ob...        12.3       1526.0   
1  Percent of students in grades 9-12 who have an...        15.1       1526.0   
2                                                NaN         NaN          NaN   

   Total                    GeoLocation StratificationID1  
0  Total  (32.840571122, -86.631860762)           OVERALL  
1  Total  (32.840571122, -86.631860762)           OVERALL  
2    NaN                            NaN               NaN

In [32]:
access_obesity = access_obesity.drop(access_obesity.columns[[2, 3, 4, 6, 7, 11, 13]], axis = 1)

In [54]:
access_obesity

Year          State  Individuals 3 - 17  \
0    2001        Alabama              372391   
1    2001        Alabama              372391   
2    2001         Alaska               83774   
3    2001        Arizona              586415   
4    2001       Arkansas              227795   
..    ...            ...                 ...   
573  2012  West Virginia              269136   
574  2012      Wisconsin              957052   
575  2012      Wisconsin              957052   
576  2012        Wyoming               96979   
577  2012        Wyoming               96979   

                                              Question  Data_Value  \
0    Percent of students in grades 9-12 who have ob...        12.3   
1    Percent of students in grades 9-12 who have an...        15.1   
2                                                  NaN         NaN   
3                                                  NaN         NaN   
4    Percent of students in grades 9-12 who have an...        15.7   
..                                                 ...         ...   
573  Percent of students in grades 9-12 who have an...        15.5   
574  Percent of students in grades 9-12 who have an...        13.0   
575  Percent of students in grades 9-12 who have ob...        11.6   
576  Percent of students in grades 9-12 who have ob...        10.7   
577  Percent of students in grades 9-12 who have an...        12.8   

     Sample_Size                     GeoLocation  
0         1526.0   (32.840571122, -86.631860762)  
1         1526.0   (32.840571122, -86.631860762)  
2            NaN                             NaN  
3            NaN                             NaN  
4         1586.0   (34.748650124, -92.274490743)  
..           ...                             ...  
573       1561.0   (38.665510202, -80.712640135)  
574       2771.0   (44.393191174, -89.816370742)  
575       2771.0   (44.393191174, -89.816370742)  
576       2910.0  (43.235541343, -108.109830353)  
577       2910.0  (43.235541343, -108.109830353)  

[578 rows x 7 columns]

In [67]:
test = access_obesity.dropna()
test = test[test.Year == 2012]
len(test.State.unique())
#amount of states accounted for by year
#2001:21 , 2003:31 , 2007:41 , 2009:42 , 2011:44 , 2012: 42

42

In [72]:
test = access_obesity[access_obesity.Question.isna()]
test[test['Year']==2012]

Year                 State  Individuals 3 - 17 Question  Data_Value  \
493  2012            California             6297415      NaN         NaN   
494  2012              Colorado              902808      NaN         NaN   
499  2012  District of Columbia              100874      NaN         NaN   
510  2012               Indiana             1046635      NaN         NaN   
511  2012                  Iowa              507270      NaN         NaN   
526  2012             Minnesota              970024      NaN         NaN   
553  2012                Oregon              706225      NaN         NaN   
554  2012          Pennsylvania             2126856      NaN         NaN   
571  2012            Washington             1191561      NaN         NaN   

     Sample_Size GeoLocation  
493          NaN         NaN  
494          NaN         NaN  
499          NaN         NaN  
510          NaN         NaN  
511          NaN         NaN  
526          NaN         NaN  
553          NaN         NaN  
554          NaN         NaN  
571          NaN         NaN

In [73]:
access_obesity = access_obesity.dropna()

In [74]:
access_obesity.to_csv('cleaned_data/USA_child_internet_obesity_2001_2012.csv')

In [77]:
access_obesity.corr()

Year  Individuals 3 - 17  Data_Value  Sample_Size
Year                1.000000            0.153522    0.214450     0.106654
Individuals 3 - 17  0.153522            1.000000    0.125400     0.234209
Data_Value          0.214450            0.125400    1.000000    -0.041938
Sample_Size         0.106654            0.234209   -0.041938     1.000000

#### correlation coefficient of child internet access to childhood nutritional health problems is 0.1254